src/Tests/dotnet-new.Tests/CommonTemplatesTests.cs

```diff
@@ -162,24 +173,24 @@ public void NuGetConfigPermissions()
 
             var templatesToTest = new[]
             {
-                new { Template = consoleTemplateShortname,  Frameworks = new[] { null, "net7.0" } },
-                new { Template = "classlib", Frameworks = new[] { null, "net7.0", "netstandard2.0", "netstandard2.1" } }
+                new { Template = consoleTemplateShortname,  Frameworks = new[] { null, "net6.0", "net7.0", "net8.0" } },
+                new { Template = "classlib", Frameworks = new[] { null, "net6.0", "net7.0", "net8.0", "netstandard2.0", "netstandard2.1" } }
             };
 
             //features: top-level statements; nullables; implicit usings; filescoped namespaces
 
             string[] unsupportedLanguageVersions = { "1", "ISO-1" };
-            string[] unsupportedFrameworkVersions = { "net5.0", "netcoreapp3.1" };
-            string?[] supportedLanguageVersions = { null, "ISO-2", "2", "3", "4", "5", "6", "7", "7.1", "7.2", "7.3", "8.0", "9.0", "10.0", "11.0", "11", "latest", "latestMajor", "default", "preview" };
+            //C# 12 is not supported yet - https://github.com/dotnet/sdk/issues/29195
+            string?[] supportedLanguageVersions = { null, "ISO-2", "2", "3", "4", "5", "6", "7", "7.1", "7.2", "7.3", "8.0", "9.0", "10.0", "11.0", "11", /*"12",*/ "latest", "latestMajor", "default", "preview" };
 
-            string?[] nullableSupportedInFrameworkByDefault = { null, "net7.0", "netstandard2.1" };
-            string?[] implicitUsingsSupportedInFramework = { null, "net7.0" };
-            string?[] fileScopedNamespacesSupportedFrameworkByDefault = { null, "net7.0" };
+            string?[] nullableSupportedInFrameworkByDefault = { null, "net7.0", "net8.0", "netstandard2.1" };
+            string?[] implicitUsingsSupportedInFramework = { null, "net6.0", "net7.0", "net8.0" };
```

Comment:
given that test runtime is decreased now, i would recommend to add "netcoreapp3.1" here - it is still supported.

Reply:
Skip this, 3.1 is almost EOL.